internal/api/v1/application/part.go

```diff
@@ -118,8 +118,38 @@ func fetchAppChart(c *gin.Context, ctx context.Context, logger logr.Logger, clus
 		return apierror.InternalError(err)
 	}
 
+	logger.Info("input", "chart archive", chartArchive)
+
+	// Try to read the archive as local path first, before falling back to retrieval via http.
+
+	file, err := os.Open(chartArchive)
+	if err == nil {
+		logger.Info("input", "chart archive", "is file")
+
+		fileInfo, err := file.Stat()
+		if err == nil {
```

Comment:
Should we handle the err and return? Or we still try to retrieve it via http?

Reply:
I did not wish to use something brittle like a check for the `/assets/...` prefix in the chart archive path to distinguish between a file versus an url.

I simply test if the thing is a path, and if not I assume that it is a proper url.

I.e. for normal deployments (dev, and through chart directly) the `chartArchive` is `https://github.com/epinio/helm-charts/releases/download/epinio-application-0.1.24`. This fails the `os.Open()`, and so I go and use `http.Get()` on it.

And for the integration we have `/assets/epinio-application-0.1.24.tgz` which succeeds the `os.Open()` and thus we go and treat it as file.

If we have a file or url which is bad, i.e. does not exist it would simply fail both `os.Open()` and `http.Get()`, and we will get the error from the latter.

Reply:
Yes, I'm fine with the `os.Open()`, but if the `file.Stat()` fails then we are not returning, so we are continuing the flow and go to the `http.Get()`. Or maybe I missed something!

Reply:
Yes. I treat the `file.Stat()` failure same as an `os.Open()` failure. If we cannot stat the file, despite being able to open it I see it as `not a file` too.